tests/FsCodec.SystemTextJson.Tests/AutoUnionTests.fs

```diff
@@ -22,6 +22,10 @@ let [<Xunit.Fact>] ``Basic characteristics`` () =
     test <@ Not { body = "A"; opt = None; list = [] } = serdes.Deserialize "{\"case\":\"Not\",\"body\":\"A\",\"list\":[]}" @>
     test <@ Not { body = "A"; opt = None; list = ["A"] } = serdes.Deserialize "{\"case\":\"Not\",\"body\":\"A\",\"list\":[\"A\"]}" @>
 
+let [<Xunit.Fact>] ``It throws on null strings`` () =
+    raises <@ serdes.Deserialize<string> "null" @>
+    raises <@ serdes.Deserialize<NotAUnion> "{\"body\": null, \"opt\": null, \"list\": []}" @>
```

Comment:
How can one roundtrip and/or safely handle `null` values if this throws?
Need to have a way to handle `null` values in a blob of JSON without having to drop to low level STJ overrides

Reply:
Options are supported. This forces you to make the nullability explicit. Though you might be asking for something different?

Reply:
ok but given opt is a `string option`, I would expect the `null` to then come back as either `None` or `Some null`, whereas this suggests it will throw when presented with that ?

More importantly, if I had emitted a `None` via `string option`, would I be able to load it from whatever that would have rendered as ?

Reply:
Ah, the error is because the body is null. The opt deserializes fine

Reply:
Ah, that makes sense. (Though I guess that's a hint that having a specifically typed exception with a message that pinpoints the issue to `null` not being allowed for `string` _unless wrapped in an Option_ will make this the best experience for when people inevitably run into it.)

Reply:
I've changed the message to indicate what needs to be done to remedy it.